### PR TITLE
Add notifications on http request

### LINKED ISSUE #98: Notify the user on API errors
## Description

**As a** a user,

**When** I perform an action that require an API call

**And** The API call fails

**I want** to be notified

**so that** I know what went wrong

### Scenario 1

 **Scenario** | Create or delete tag
 ------------ | -------------------------------------------------
 **Given**    | The tag update panel
 **When**     | I create or delete a tag
**And**        | The related API call is in error
 **Then**     | I want to receive a notification

### Constraints

There are many other cases where API errors are not catched.

### PR DESCRIPTION
**Pull request by BrahimS**

### CODE REVIEW

File: frontend/src/composables/WorkSpace/useDomainData.js [hunk 1]
```diff
@@ -98,8 +98,19 @@ export default function useDomainData() {
...
-			refreshDomainTag(choosenNodeID.value);
+			await refreshDomainTag(choosenNodeID.value);
+			quasar.notify({
+				type: "positive",
+				message: "Tag has been successfully added",
+			});
+		} catch (error) {
+			quasar.notify({
+				type: "negative",
+				message: "Sorry, tag has not been added",
```
Comment (Mari0nV) [inline frontend/src/composables/WorkSpace/useDomainData.js]: Maybe add an explanation for the user : like "Error while adding tag : <reason>"
the reason being the response from the API (parsed of course)

File: frontend/src/composables/WorkSpace/useDomainData.js [hunk 2]
```diff
@@ -141,28 +152,40 @@ export default function useDomainData() {
...
+			await API.delete(`/domain/${choosenNodeID.value}/tag/${id}`);
+			await refreshDomainTag(choosenNodeID.value);
+			quasar.notify({
+				type: "positive",
+				message: "Tag has been successfully deleted",
+			});
+		} catch (error) {
+			quasar.notify({
+				type: "negative",
+				message: "Sorry, tag has not been deleted",
```
Comment (Mari0nV) [inline frontend/src/composables/WorkSpace/useDomainData.js]: Same comment as above

**Comment by Mari0nV** [comment]
In the US related to this PR, I mentioned the creation and deletion of tags as an example of API calls that were not catched, but aren't there other API calls that are not catched in the code?
The purpose of this task was to catch them all.

**Review by Mari0nV — CHANGES_REQUESTED** [review]
- Improve error message with more details of the error
- Find other occurrences of API calls that are not catched

File: frontend/src/composables/WorkSpace/useDomainData.js (hunk 1 above)
Comment (BrahimS) [reply frontend/src/composables/WorkSpace/useDomainData.js]: The Error message should be humain readable. but Error while adding tag works as well :)

File: frontend/src/composables/WorkSpace/useDomainData.js (hunk 2 above)
Comment (BrahimS) [reply frontend/src/composables/WorkSpace/useDomainData.js]: The Error message should be humain readable. but Error whitle adding tag works as well :)

File: frontend/src/composables/WorkSpace/useDomainData.js [hunk 3]
```diff
@@ -98,8 +98,19 @@ export default function useDomainData() {
...
-			refreshDomainTag(choosenNodeID.value);
+			await refreshDomainTag(choosenNodeID.value);
+			quasar.notify({
+				type: "positive",
+				message: "Tag has been successfully added",
+			});
+		} catch (error) {
+			quasar.notify({
+				type: "negative",
+				message: "Error whitle adding tag",
```
Comment (Mari0nV) [inline frontend/src/composables/WorkSpace/useDomainData.js]: there is a typo  (and same line 165)
whitle -> while

Comment (BrahimS) [reply frontend/src/composables/WorkSpace/useDomainData.js]: ok :)

**Review by Mari0nV — APPROVED** [review]
PR approved but we need better error messages for the user. I'll create another issue about that

**Pull request merged**